hbase-migration-tools/bigtable-hbase-replication/bigtable-hbase-2.x-replication/src/test/java/com/google/cloud/bigtable/hbase2_x/replication/HbaseToCloudBigtableReplicationEndpointTest.java

```diff
@@ -425,4 +411,78 @@ public void testWriteFailureToBigtableDoesNotStallReplication()
         "Value mismatch", TestUtils.getValue(0), CellUtil.cloneValue(actualCells.get(1)));
     Assert.assertEquals(0, actualCells.get(1).getTimestamp());
   }
+
+  @Test
+  public void testMutationReplicationWithWALEntryFilter() throws IOException, InterruptedException {
+    Put put1 = new Put(FILTERED_ROW_KEY);
+    put1.addColumn(TestUtils.CF1, TestUtils.COL_QUALIFIER, 0, FILTERED_ROW_KEY);
+    put1.addColumn(TestUtils.CF2, TestUtils.COL_QUALIFIER, 0, FILTERED_ROW_KEY);
+    hbaseTable.put(put1);
+
+    Put put2 = new Put(ROW_KEY);
+    put2.addColumn(TestUtils.CF1, TestUtils.COL_QUALIFIER, 0, ROW_KEY);
+    put2.addColumn(TestUtils.CF2, TestUtils.COL_QUALIFIER, 0, ROW_KEY);
+    hbaseTable.put(put2);
+
+    TestUtils.waitForReplication(
+        () -> {
+          //  replicate Entries will be zero
+          return TestReplicationEndpoint.replicatedEntries.get() >= 1;
```

Comment:
2

Reply:
It should be 1 as FILTERED_ROW_KEY is not passed in replicateContext.